### PR TITLE
NVRTC compilation flag update

### PR DESCRIPTION
**Pull request by jjsjann123**
Allow binary dump when compiling to sass
Skip assertion for kernel code in release build
Dump ptxas register usage for sass kernel in debug build

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/executor_utils.cpp [hunk 1]
```diff
@@ -678,9 +678,15 @@ NvrtcFunction nvrtcCompile(
...
-  // Add line info to generated kernels
 #ifndef NDEBUG
+  // Add line info to generated kernels
   args.push_back("-lineinfo");
+  // Add register usage in compilation
+  args.push_back("--ptxas-options");
+  args.push_back("--verbose");
+#else
+  // Avoid excessive register usage from assertion
+  args.push_back("-DNDEBUG");
```
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/executor_utils.cpp:686]: Per @naoyam's comment

File: torch/csrc/jit/codegen/cuda/executor_utils.cpp [hunk 2]
```diff
@@ -678,9 +678,15 @@ NvrtcFunction nvrtcCompile(
 #endif
   }
 
-  // Add line info to generated kernels
 #ifndef NDEBUG
```
Comment (csarofeen) [inline torch/csrc/jit/codegen/cuda/executor_utils.cpp:681]: Should we have an env based option as well?

File: torch/csrc/jit/codegen/cuda/executor_utils.cpp [hunk 3]
```diff
@@ -750,6 +756,16 @@ NvrtcFunction nvrtcCompile(
 
       TORCH_INTERNAL_ASSERT(
           false, code.c_str(), "\nCUDA NVRTC compile error: ", log.data());
+#ifndef NDEBUG
+    } else {
```
Comment (csarofeen) [inline torch/csrc/jit/codegen/cuda/executor_utils.cpp]: This one seems like it should be an ENV variable.

**Review by csarofeen — APPROVED** [review]
LGTM thanks for the changes.

File: torch/csrc/jit/codegen/cuda/executor_utils.cpp (hunk 2 above)
Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/executor_utils.cpp:681]: I'll add this to `PYTORCH_NVFUSER_DUMP`.

**Pull request merged**